### DISCUSSION TITLE
Support for direct PMTiles generation

### DISCUSSION THREAD

**Discussion by danvinci** [maplibre/martin#1213] [Q&A]
Hi folks!

I'm new to GIS and trying to find my way around the pieces of the stack and how they fit together.

I'm pretty set on using PMTiles in my project, and thinking of serving them from a CDN as described in this 
[blog article](https://www.openstreetmap.org/user/ZeLonewolf/diary/402227) - for then consuming them in the browser through MapLibre GL.

As part of the building/learning journey, I want to generate the tiles myself, starting from OSM data. 
[Planetiler](https://github.com/onthegomap/planetiler) seems the obvious choice, but wondering if Martin could fit this task as well.

I see that the Martin [book](https://maplibre.org/martin/tools.html) mentions several utilities for bulk tiles generation, although it seem they can only output MBTiles.

I have some questions:
- When using the utilities, OSM data should be loaded into a PostGIS instance for Martin to read from for generation - correct? Or direct consumption of PBF files is possible without involving a data loader and running a DB?
- Is there an utility for bulk tiles generation directly to PMTiles?

If not, I wonder if the workflow of loading OSM data & running PostGIS -> MBTiles generation -> conversion to PMTiles could somehow beat direct OSM -> PMTiles generation offered by Planetiler in terms of time/performance - given the same machine.

The use case would be to build/re-build tiles for the whole planet ~once/week.

Curious to see if anyone has ever considered this, as well as if there's something off in my current understanding.

**Comment by nyurik** [comment]
[Planetiler](https://github.com/onthegomap/planetiler) supports `.pmtiles` output. If your data doesn't change much, and the same for everyone, that would be the preferred choice. HOW you serve it is separate though - you could use direct http requests, but some people don't like that for various security and potential abuse issues, so they prefer to use other options like Martin to do the "real server" serving. Martin can serve pmtiles too, even as a labmda (this was just merged, not yet released, but soon).  If you are serving real time or custom data from a DB, something like Martin is a must though.
